apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts

```diff
@@ -0,0 +1,339 @@
+                if (this.taxesList.length === 1) {
+                    this.getFormControl('taxNumber').patchValue(this.taxesList[0].value);
+                }
+                if (this.isCompanyMode) {
+                    this.hasTaxNumber = true;
+                }
+                this.getLiabilitiesPayment();
+            }
+        });
+    }
+     /**
+     * Get Current company branches information
+     *
+     * @private
+     * @memberof VatLiabilitiesPayments
+     */
+    private getCurrentCompanyBranchTaxNumber(): void {
+            this.currentCompanyBranches$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                const currentBranchUniqueName = this.generalService.currentBranchUniqueName;
+                this.currentBranch = cloneDeep(response.find(branch => branch?.uniqueName === currentBranchUniqueName));
+                this.hasTaxNumber = this.currentBranch?.addresses?.filter(address => address?.taxNumber?.length > 0)?.length > 0;
+                if (this.hasTaxNumber) {
+                    this.loadTaxDetails();
+                }
+            }
+        });
+    }
+
+    /**
+    * VAT Liabilities/Payments API Call
+    *
+    * @memberof VatLiabilitiesPayments
+    */
+    public getLiabilitiesPayment(): void {
+        let payload = this.generalService.getUserAgentData();
+        payload["Gov-Test-Scenario"] = "MULTIPLE_PAYMENTS_2018_19";
+        this.componentStore.getLiabilityPaymentList({ payload: payload, searchForm: this.searchForm.value, isPaymentMode: this.isPaymentMode });
+    }
+
+    /**
+    * This will use for init main formgroup
+    *
+    * @private
+    * @memberof VatLiabilitiesPayments
+    */
+    private initVatLiabilityPaymentForm(): void {
+        this.searchForm = this.formBuilder.group({
+            companyUniqueName: [""],
+            branchUniqueName: [''],
+            taxNumber: [''],
+            from: [''],
+            to: ['']
+        });
+    }
+
+    /**
+    * Handle Dropdown callback for Tax Number and save value to form
+    *
+    * @param {*} event
+    * @memberof VatLiabilitiesPayments
+    */
+    public taxNumberSelected(event: any): void {
+        if (event?.value) {
+            this.getFormControl('taxNumber').patchValue(event.value);
+        }
+    }
+
+    /**
+    * Handle Dropdown callback for Branch and save value to form
+    *
+    * @param {*} event
+    * @memberof VatLiabilitiesPayments
+    */
+    public branchSelected(event: any): void {
+        if (event?.value) {
+            this.getFormControl('branchUniqueName').patchValue(event.value);
+        }
+    }
+
+    /**
+    * Get Universal Date Observable from Store and subscribed
+    *
+    * @private
+    * @memberof VatLiabilitiesPayments
+    */
+    private getUniversalDatePickerDate(): void {
+        this.store.pipe(select(stateStore => stateStore.session.applicationDate), takeUntil(this.destroyed$)).subscribe((dateObj) => {
```

Comment:
don't directly use global store if we have component store

Reply:
Done